applications/RANSApplication/tests/cpp/test_vms_monolithic_k_based_condition.cpp

```diff
@@ -0,0 +1,345 @@
+            KRATOS_CHECK_EQUAL(dofs[local_index++]->EquationId(),
+                               r_geometry[i_node].Id() * 4);
+            KRATOS_CHECK_EQUAL(dofs[local_index]->GetVariable(), VELOCITY_Y);
+            KRATOS_CHECK_EQUAL(dofs[local_index++]->EquationId(),
+                               r_geometry[i_node].Id() * 4 + 1);
+            KRATOS_CHECK_EQUAL(dofs[local_index]->GetVariable(), PRESSURE);
+            KRATOS_CHECK_EQUAL(dofs[local_index++]->EquationId(),
+                               r_geometry[i_node].Id() * 4 + 3);
+        }
+    }
+}
+
+KRATOS_TEST_CASE_IN_SUITE(RansVMSMonolithicKBasedWall2D2N_CalculateLocalSystem, KratosRansFastSuite)
+{
+    // Setup:
+    Model model;
+    auto& r_model_part = RansVMSMonolithicKBasedWall2D2N_SetUp(model);
+
+    // Test:
+    Matrix LHS, ref_LHS;
+    Vector RHS, ref_RHS;
+    auto& r_condition = r_model_part.Conditions().front();
+
+    // checking for no-wall function
+    r_condition.SetValue(RANS_IS_WALL_FUNCTION_ACTIVE, 0);
+    r_condition.Initialize();
+    r_condition.CalculateLocalSystem(
+        LHS, RHS, static_cast<const ProcessInfo>(r_model_part.GetProcessInfo()));
+
+    // setting reference values
+    ref_RHS = ZeroVector(6);
+    ref_LHS = ZeroMatrix(6, 6);
+
+    KRATOS_CHECK_VECTOR_NEAR(RHS, ref_RHS, 1e-12);
+    KRATOS_CHECK_MATRIX_NEAR(LHS, ref_LHS, 1e-12);
+
+    // checking for wall function
+    r_condition.SetValue(RANS_IS_WALL_FUNCTION_ACTIVE, 1);
+    r_condition.Initialize();
+    r_condition.CalculateLocalSystem(
+        LHS, RHS, static_cast<const ProcessInfo>(r_model_part.GetProcessInfo()));
+
+    // setting reference values
+    ref_RHS = ZeroVector(6);
+    ref_LHS = ZeroMatrix(6, 6);
+
+    KRATOS_CHECK_VECTOR_NEAR(RHS, ref_RHS, 1e-12);
+    KRATOS_CHECK_MATRIX_NEAR(LHS, ref_LHS, 1e-12);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(RansVMSMonolithicKBasedWall2D2N_CalculateLeftHandSide, KratosRansFastSuite)
+{
+    // Setup:
+    Model model;
+    auto& r_model_part = RansVMSMonolithicKBasedWall2D2N_SetUp(model);
+
+    // Test:
+    Matrix LHS, ref_LHS;
+    auto& r_condition = r_model_part.Conditions().front();
+
+    // checking for no-wall function
+    r_condition.SetValue(RANS_IS_WALL_FUNCTION_ACTIVE, 0);
+    r_condition.Initialize();
+    r_condition.CalculateLeftHandSide(
+        LHS, static_cast<const ProcessInfo>(r_model_part.GetProcessInfo()));
+
+    // setting reference values
+    ref_LHS = ZeroMatrix(6, 6);
+
+    KRATOS_CHECK_MATRIX_NEAR(LHS, ref_LHS, 1e-12);
+
+    // checking for wall function
+    r_condition.SetValue(RANS_IS_WALL_FUNCTION_ACTIVE, 1);
+    r_condition.Initialize();
+    r_condition.CalculateLeftHandSide(
+        LHS, static_cast<const ProcessInfo>(r_model_part.GetProcessInfo()));
+
+    // setting reference values
+    ref_LHS = ZeroMatrix(6, 6);
+
+    KRATOS_CHECK_MATRIX_NEAR(LHS, ref_LHS, 1e-12);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(RansVMSMonolithicKBasedWall2D2N_CalculateRightHandSide, KratosRansFastSuite)
```

Comment:
Convention.

Reply:
This I would like to keep as it is because its a long name, having underscores gives little bit of readability :)